funding/manager.go

```diff
@@ -2860,14 +2856,9 @@ func (f *Manager) sendFundingLocked(completeChan *channeldb.OpenChannel,
 	// send fundingLocked until we succeed, or the fundingManager is shut
 	// down.
 	for {
-		connected := make(chan lnpeer.Peer, 1)
-		f.cfg.NotifyWhenOnline(peerKey, connected)
-
-		var peer lnpeer.Peer
-		select {
-		case peer = <-connected:
-		case <-f.quit:
-			return ErrFundingManagerShuttingDown
+		peer, err := f.waitForPeerOnline(completeChan.IdentityPub)
```

Comment:
Previously we used `peerKey`, but now it's `completeChan.IdentifyPub`. I assume they are always the same? A subtle difference is, previously we'd always use the bytes copied above in this `for` loop, now we are accessing it via the pointer. Not familiar with the callpath, just wanna be sure that the value pointed by `completeChan.IdentityPub` always stays the same.

Reply:
Cool thanks for the explanation. Yeah imo simplicity is always preferred.